src/Microsoft.Extensions.DiagnosticAdapter/DiagnosticSourceAdapter.cs

```diff
@@ -130,16 +176,23 @@ public void Write(string diagnosticName, object parameters)
 
         private class Listener
         {
-
             public Listener(object target, Func<string, bool> isEnabled)
             {
                 Target = target;
+                IsEnabled = (isEnabled == null) ? (Func<string, object, object, bool>) null : (a, b, c) => isEnabled(a);
```

Comment:
Yeah you figured it out 👍 - flow this up one level of abstraction and that will dedupe some code